### ISSUE TITLE
Headless chrome issue with custom rootElement

### ISSUE THREAD

**Issue by segurev** [angular/protractor#5055]
Bug report
---
### Use case: 
- Protractor E2E tests against a local AngularJs (1.5) app
### Background:
- AngularJS app uses custom bootstrapping
### My issue:
- **Test execution works fine in Chrome but not in Headless Chrome**
  - In Headless Chrome there are irregular errors for some tests
    - `Failed: unknown error: Cannot read property '$$testability' of undefined`
    - `Failed: unknown error: [ng:btstrpd] App already bootstrapped with this element 'document'`
### Tested approaches (all worked with Chrome but not with Headless Chrome)
- Different "rootElement" configurations in protractor.conf.js:
  - No rootElement (use default)
  - Set rootElement to [module="moduleName"]
  - Set rootElement to body
### Protractor Chrome config:
- `    multiCapabilities: [
        {
            browserName: 'chrome',
            chromeOptions: {
                args: ['--headless', '--disable-gpu', '--window-size=1044,900', '--no-sandbox', '--display=:0', '--proxy-server="direct://"', '--proxy-bypass-list=*']
            }
        }
    ]`
### Environment
- Node Version: `v8.11.2`
- Protractor Version: `5.3.2`
- Angular Version: `1.5.11`
- Browser(s): `(Headless) Chrome (70.0.3538.110)`
  - Chrome Driver: `2.36`
- Operating System and Version: `Windows 10 Enterprise`

**Comment by cnishina** [comment]
Per your question, it looks like this only happens on headless? Does it work when the browser is not headless?

**Comment by jwedel** [comment]
@cnishina Yes, it works in non-headless mode.

**Comment by segurev** [comment]
@cnishina:
yes, it only happens with headless. In normal Chrome everything seems fine.

**Comment by segurev** [comment]
@cnishina 
We would really appreciate help with that one. Do you need any more information?

**Comment by segurev** [comment]
@cnishina 
We would really appreciate help with that one. Do you need any more information?

**Comment by fbutter** [comment]
@cnishina any news on this one?

**Comment by Nas3nmann** [comment]
@cnishina same issue here. Working with normal chrome :(

**Comment by Nas3nmann** [comment]
FYI: I found a quite useful workaround here: http://www.sebdangerfield.me.uk/2014/01/angularjs-protractor-app-already-bootstrapped-error/
No errors so far since we're using this.
